students/mt-desta/lab1/README.md

```diff
@@ -0,0 +1,91 @@
+
+### **Lab #1 Clustering**
+
+#### <p style='text-aligh: justify'> 1. For this lab work I have selected the [IRIS Dataset](https://www.kaggle.com/datasets/arshid/iris-flower-dataset) and [Mall Customers](https://www.kaggle.com/datasets/vjchoudhary7/customer-segmentation-tutorial-in-python/data) datasets from kaggle. </p>
+####   <p style='text-aligh: justify'> 1.1 Since both datasets didn't have class labels there was no need to perform the first step in the given tasks. Although FinalGrade from the Student Performance can be considered as a class label I decided to treat it as a feature. </p>
+
+#### 1.2 I have used scatter plots of the given datasets to visualise the data in the datasets. I use the [importing_data.py](./source/importing_data.py) to import and visualise the data in one go.
+
+<img src="img/image1.png">
+
+####  <p style='text-aligh: justify'> 1.3 The first dataset is made up of **clusters with jumpers**. As clearly seen on the plot we have one or two points that connect the areas with dense consentration of points. As for the second dataset, it has no clear shape but I assume is made up of **overlapping clusters**.  </p>
+
+#### <p style='text-aligh: justify'> 1.4 Though it is not clear, for the first dataset my assumption is at least 5 clusters by looking at the graph. As for the second dataset, there seem to be 2 or 3 areas with consentration but I cannot be so sure.</p>
+
+#### 2. I have implented a hierarchical algorithm based on Agglomerative clustering. [my_hierarchy.py](./source/my_hierarchy.py) contains the code for this implementation. It takes the number of desired clusters, linkage method and a dataframe as arguements. It outputs the linkage matrix and labels. I have implemented a function tha takes these and plots a dendrogram.
+
+<img src="img/image3.png">
+<img src="img/image4.png">
+
+#### 2.1 Dendrograms for the two datasets
+
+<img src="img/image5.png">
+        
+#### 2.2 Optimum number of Clusters
```

Comment:
Calculate the optimal number of clusters based on the maximum increment R_t for hierarchical clustering based on Ward distance

Reply:
resolved issue.